### ISSUE TITLE
Get TypeRepr/Type type params & substitute

### ISSUE THREAD

**Issue by cchantep** [lampepfl/dotty#12392]
In macro/quoted, doesn't seem possible to neither get the type parameters for a `Type`/`TypeRepr`/`Symbol`, nor `subtituteTypes`.

**Comment by nicolasstucki** [comment]
Could you elaborate?

**Comment by cchantep** [comment]
Scala2 `Symbol` provides `typeParams` and a `subsituteTypes` function.

**Comment by nicolasstucki** [comment]
Could you point me to the documentation (or code) of those methods?

**Comment by cchantep** [comment]
See the description

**Comment by nicolasstucki** [comment]
For `subtituteTypes`, it seems that we need to expose `instantiateParamInfos` and `instantiate` on `LambdaType`.

Not sure what `typeParams` returns. Is it the type bounds of the parameters?

**Comment by cchantep** [comment]
I don't think that `subtituteTypes` must be limited to `LambdaType` but on all `Type` (or `TypeRepr`).

`typeParams` returns the applied or unapplied type parameters.

**Comment by cchantep** [comment]
Any news?

**Comment by cchantep** [comment]
up ?

**Comment by cchantep** [comment]
up ?

**Comment by cchantep** [comment]
Any feeback? From my point of view, that's a blocker with no pragmatic alternative.

**Comment by nicolasstucki** [comment]
See #13166. Thought I'm lacking a use-case to put as a test.
